### PR TITLE
Issue #243 two versions of SORT program

### PR DESCRIPTION
**Pull request by abekornelis**
Issue #243 two versions of SORT program
build and test procedures modified to use the remaining source version in linklib directory.

### CODE REVIEW

File: sort/RUNSORT.BAT [hunk 1]
```diff
@@ -1,42 +1,79 @@
...
+IF ERRORLEVEL 0 (echo **** No errors in TESTSRT4 verify TESTSRT3.OUT
+         ) ELSE (echo **** ++ Error! in TESTSRT4 verify TESTSRT3.OUT
+                 set  z_ReturnCode=8)
+rem
+if exist SORTWK*.TMP erase SORTWK*.TMP
+if %z_ReturnCode% == 0 echo **** All SORT tests completed okay
 
+:return
+rem ----- Lvl(%z_NestLevel%)  End %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
+exit /b %z_ReturnCode%
```
Comment (jyganci) [inline sort/RUNSORT.BAT]: sort/RUNSORT.BAT is obsolete. Replaced by rt/bat/RUNSORT.BAT and rt/bash/runsort. Make changes there and delete sort/RUNSORT.BAT.

Comment (abekornelis) [reply sort/RUNSORT.BAT]: Okay, I'll look into it. Thanks for notifying me - I had not noticed.

File: rt/bash/runsort [hunk 2]
```diff
@@ -6,8 +6,8 @@ cd $(dirname $0)    # set to directory of script - rt/bash
 cd ..               # up to rt
 cd ..               # up to z390
 
-# rebuild sort/SORT, sort/TESTSRT[1-4]
-bash/asml sort/SORT $1 $2 $3 $4 $5
+# copy sort loadmod, assemble test programs
+cp linklib/SORT.390 sort/SORT.390
```
Comment (jyganci) [inline rt/bash/runsort]: Okay, but not what I would do. Sending you via email an alternative version of rt/bash/runsort.

File: sort/RUNSORT.BAT (hunk 1 above)
Comment (jyganci) [reply sort/RUNSORT.BAT]: It appears that you've decided to keep the obsolete sort/RUNSORT.BAT. Why?

File: z390test/src/test/groovy/org/z390/test/RunSortTests.groovy [hunk 3]
```diff
@@ -16,18 +16,9 @@ class RunSortTests extends z390Test {
     ]
 
     var runOptions = [
-            "SYS390(${basePath('sort')})"
+            "SYS390(${basePath('linklib')})"
```
Comment (jyganci) [inline z390test/src/test/groovy/org/z390/test/RunSortTests.groovy:19]: Have you run the groovy script?

File: rt/bat/RUNSORT.BAT [hunk 4]
```diff
@@ -13,80 +13,122 @@ rem ----- Lvl(%z_NestLevel%) Start %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
...
+call bat\ASML %z_TraceMode% %sortlib%\TESTSRT3 %1 %2 %3 %4 %5 || goto error
+call bat\ASML %z_TraceMode% %sortlib%\TESTSRT4 %1 %2 %3 %4 %5 || goto error
+
+rem Test 1: TESTSORT = sort ASCII 21 record simple sort
+set SORTIN=%sortlib%\TESTSORT.IN[RECFM=FT,LRECL=80]
+set SORTOUT=%sortlib%\TESTSORT.OUT
+set SYSIN=%sortlib%\TESTSORT.INI
+if exist %SORTOUT% erase %SORTOUT%
+set SORTOUT=%SORTOUT%[RECFM=FT]
+rem use bash/ez390 instead of bash/exec so SORT.ERR, SORT.LOG,
```
Comment (jyganci) [inline rt/bat/RUNSORT.BAT]: You might change "bash/exec" to "bat\EXEC" and change "bash/ez390" to bat\ez390".

File: rt/bash/runsort [hunk 5]
```diff
@@ -114,6 +111,9 @@ fi
 
 # done with tests
 
+# remove residual files from SORT executions
+rm -f sort/SORT.ERR sort/SORT.LOG sort/SORT.STA
```
Comment (jyganci) [inline rt/bash/runsort:115]: You added a line in RUNSORT.BAT that deletes SORTWK*.TMP. Please add it here. Here's the line to add:
```
rm -f SORTWK*.TMP
```

**Review by jyganci — APPROVED** [review]
Looks good. I did not read the entire updated files; just (1) the one line addition to rt/bashrunsort and (2) the one linc change to rt/bat/RUNSORT.BAT.

**Pull request merged**